src/containers/Application/ApplicationPageWrapper.tsx

```diff
@@ -44,10 +57,42 @@ const ApplicationPageWrapper: React.FC = () => {
         push,
         replace,
       })
+
+      if (sectionCode && page)
+        setCurrentSection(templateSections.find(({ code }) => code === sectionCode))
+    }
+  }, [isApplicationLoaded, sectionCode, page])
+
+  useEffect(() => {
+    const progressStructure = buildProgressInApplication({
```

Comment:
I think we should also check for responsesLoading and return if it's true, (would need to add it to watched variables array)